File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/AppRequestedEvent.java
package work.socialhub.kslack.entity.event;

import com.github.seratch.jslack.api.model.admin.AppRequest;

/**
 * https://api.slack.com/events/app_requested
 */
public class AppRequestedEvent implements Event {

    public static final String TYPE_NAME = "app_requested";

    private final String type = TYPE_NAME;
    private AppRequest appRequest;

    public String getType() {
        return this.type;
    }

    public AppRequest getAppRequest() {
        return this.appRequest;
    }

    public void setAppRequest(AppRequest appRequest) {
        this.appRequest = appRequest;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/AppRequestedEvent.kt
package work.socialhub.kslack.entity.event

import work.socialhub.kslack.entity.admin.AppRequest

/**
 * https://api.slack.com/events/app_requested
 */
class AppRequestedEvent : Event {
    override val type: String = TYPE_NAME
    var appRequest: AppRequest? = null

    companion object {
        const val TYPE_NAME: String = "app_requested"
    }
}